src/Npgsql/NpgsqlConnection.cs

```diff
@@ -180,19 +181,29 @@ void GetPoolAndSettings()
             Settings = settings;
 
             var hostsSeparator = settings.Host?.IndexOf(',');
-            if (hostsSeparator.HasValue && hostsSeparator == -1)
+            if (hostsSeparator is -1)
             {
                 if (settings.TargetSessionAttributesParsed is not null &&
                     settings.TargetSessionAttributesParsed != TargetSessionAttributes.Any)
                 {
                     throw new NotSupportedException("Target Session Attributes other then Any is only supported with multiple hosts");
                 }
 
-                var portSeparator = settings.Host!.IndexOf(':');
-                if (!Path.IsPathRooted(settings.Host) && portSeparator != -1)
+                var portSeparator = settings.Host!.LastIndexOf(':');
```

Comment:
Looks duplicated with MultiHostConnectorPool... Maybe DRY this (e.g. to a static internal method on PGUtil or NpgsqlConnectionStringBuilder)?